app/utils.py

```diff
@@ -1,11 +1,20 @@
 from flask import url_for as base_url_for
 from flask import abort, request
 from six import iteritems, string_types
+import random
 from werkzeug.exceptions import BadRequest
 
 from .validation import validate_updater_json_or_400
 
 
+def random_positive_external_id():
+    """
+    Generate a random ID in the range 2^60 to 2^63-1. This will generate integers that are a consistent 19 digits long.
```

Comment:
I was wondering about whether it would all be nicer if we could base64 encode the new identifiers. Apparently that may cause problems with our analytics, which assume the id's are int's... but it'd both make our URLs shorter and also make it [even] more obvious which kind of id is in play.

Reply:
@idavidmcdonald 10^19-1 is 64-bits long, which would roll over into a negative integer ID when stored in postgres. We could do do 10**18 to 2**63-1, but I think that's probably just as opaque.

I'm open to base32/64 encoding them, but it would delay this further as we'd need to get analytics fixed up before deploying this, or accept the loss of data. Currently we can only handle integers in our analytics funnels.

I agree it would make the ID type more immediately obviously, but also the ID length is a fairly good indicator.

Open to being convinced/encouraged to go that way if people feel strongly.

Reply:
base64 is horrible to work with. base32 nice.

Reply:
If we don't go down the alphanumeric route (which I would suggest we don't or are at least somewhat blocked on this given that Ash is away), then I think we should go for 15 digits to match the service IDs in length.

Reply:
@TheDoubleK Switched to generating 15-digit IDs. Good spot. Didn't think about how they look in spreadsheets.

@idavidmcdonald Given they're now fully containable within a BIGINT, I've switched to using `.randint(10 ** 14, (10 ** 15) - 1)`

@risicle I've updated to use `random.SystemRandom`; thanks for the explanation.

Reply:
Y'know if you wanted to restrict the range to a fixed-decimal length, you could use a range up to `2**49` and stuff the most significant bits with 1s to ensure you never have the "leading zeros" ambiguity.

Reply:
(so, most significant *bit* i guess)